### ISSUE TITLE
Some questions about the codes

### ISSUE THREAD

**Issue by dgliu** [kailiang-zhong/DESCN#1]
Hi, thanks for the open source. 

But I have some questions about the codes:

1) On line 659 in the **main.py** file, the total loss includes 8 sub-losses. But **tr_loss** and **cr_loss** are removed explicitly in Equation 7 of the paper, and the CFR regularization term (i.e., **imb_dist_loss**) is not used as described in Section 3.3.

2) In the parameter configuration file (i.e., **conf_lzd_real_bin_v0_full.yaml**), I find that the weight parameters corresponding to the proposed 5 new sub-losses are all set to 0 (i.e., **prpsy_w**, **escvr1_w**, **escvr0_w**, **mu0hat_w** and **mu1hat_w**).

It may be necessary to double check whether it is consistent with the best experiment.

Best regards

**Comment by kailiang-zhong** [comment]
Hi dgliu,
Thank you for your interest in our work.

#### For Q1:
The codes determines whether to enable sub-loss by setting the weight to 0 or not. Different models will enable different sub-loss.
For example, `imb_dist_loss` is no need for TarNet, so the corresponding weight item `imb_dist_w` is set to 0, which can be seen in `./conf4models/lzd_real_data/TARNet.txt`.  But `imb_dist_w` is set to a non-zero value for CFR, which is shown in `./conf4models/lzd_real_data/ES_CFRwass.txt`.

#### ForQ2:
On line 858 in `main.py` ,
>  #./conf just contain a configuration template as default for configuration, which needs to be overwritten(key-value wise) by the hyper-parameters in ./conf4models

**The parameters of each model are actually saved in `./conf4models/`**

We load the configuration information by executing `search_params.py`
, and build the configuration terms into command-line parameters, which shown on line 106 in `search_params.py`. With [Hydra's](https://hydra.cc/) features, the configuration terms can actually work.

As described in `README`:
>A convenient way to call training and evaluation code serially:
>`python search_params.py main.py eval4real_data.py ./conf4models/lzd_real_data/DESCN.txt 1 {path_to_train_npz} 
 {path_to_test_npz}`
>More examples can be referred to code in DeepModels_real_data.ipynb

**Comment by kailiang-zhong** [comment]
@dgliu 
Hi dgliu,
Thank you for your interest in our work.

#### For Q1:
The codes determines whether to enable sub-loss by setting the weight to 0 or not. Different models will enable different sub-loss.
For example, `imb_dist_loss` is no need for TARNet, so the corresponding weight item `imb_dist_w` is set to 0, which can be seen in `./conf4models/lzd_real_data/TARNet.txt`.  But `imb_dist_w` is set to a non-zero value for CFR, which is shown in `./conf4models/lzd_real_data/ES_CFRwass.txt`.

#### ForQ2:
On line 858 in `main.py` ,
>  #./conf just contain a configuration template as default for configuration, which needs to be overwritten(key-value wise) by the hyper-parameters in ./conf4models

**The parameters of each model are actually saved in `./conf4models/`**

We load the configuration information by executing `search_params.py`
, and build the configuration terms into command-line parameters, which shown on line 106 in `search_params.py`. With [Hydra's](https://hydra.cc/) features, the configuration terms can actually work.

As described in `README`:
>A convenient way to call training and evaluation code serially:
>`python search_params.py main.py eval4real_data.py ./conf4models/lzd_real_data/DESCN.txt 1 {path_to_train_npz} 
 {path_to_test_npz}`
>More examples can be referred to code in DeepModels_real_data.ipynb
